kubernetes/e2e_test/test_client.py

```diff
@@ -117,6 +119,45 @@ def test_pod_apis(self):
 
         resp = api.delete_namespaced_pod(name=name, body={},
                                          namespace='default')
+    def test_exit_code(self):
+        client = api_client.ApiClient(configuration=self.config)
+        api = core_v1_api.CoreV1Api(client)
+
+        name = 'busybox-test-' + short_uuid()
+        pod_manifest = manifest_with_command(name, "while true;do date;sleep 5; done")
+        resp = api.create_namespaced_pod(body=pod_manifest,
+                                         namespace='default')
+        self.assertEqual(name, resp.metadata.name)
+        self.assertTrue(resp.status.phase)
+
+        while True:
+            resp = api.read_namespaced_pod(name=name,
+                                           namespace='default')
+            self.assertEqual(name, resp.metadata.name)
+            self.assertTrue(resp.status.phase)
+            if resp.status.phase == 'Running':
+                break
+            time.sleep(1)
+
+        commands_expected_values = (
+            (["false", 1]),
+            (["/bin/sh", "-c", "sleep 1; exit 3"], 3),
+            (["true", 0]),
+            (["/bin/sh", "-c", "ls /"], 0)
+        )
+        for command, value in commands_expected_values:
+            client = stream(api.connect_get_namespaced_pod_exec, name, 'default',
+                                                       command=command,
+                                                       stderr=True, stdin=False,
+                                                       stdout=True, tty=False,
+                                                       _preload_content=False)
+
+            self.assertIsNone(client.returncode)
+            client.run_forever(timeout=60)
```

Comment:
could you make the timeout smaller since the commands don't take long? maybe 10s? 

LGTM overall. Please rebase the python-base submodule after https://github.com/kubernetes-client/python-base/pull/160 is merged

Reply:
Yes, sure. I will do this 10 sec. Thanks for merging  kubernetes-client/python-base#160 is merged
 so quickly.